WORKFLOW.rst

```diff
@@ -0,0 +1,216 @@
+It is currently necessary, to change the way the RPC OpenStack team develops
+the project (i.e., adds features, fixes bugs). The old workflow is described
+in the :ref:`Old Workflow <old-workflow>` section.
+
+The new workflow is outlined under :ref:`Proposed Workflow
+<proposed-workflow>`
+
+.. _old-workflow:
+
+RPC OpenStack's Old Workflow
+============================
+
+Facts
+-----
+
+#. |rpcos| is configured with the following branches:
+
+   - ``master``
+
+   - ``kilo``
+
+   When we decide to support liberty, a third branch will appear - ``liberty``.
+
+#. Bug fixes move from ``master`` to the other branch(es) via cherry-picks.
+
+#. Bugs are hand-managed by developers and backports are tracked haphazardly.
+
+#. Developers cannot use GitHub's commit message syntax to auto-close bugs.
+
+#. ``master`` 's current focus is not well defined. (Is it the next minor
+   release of the most recent stable branch? Is it the next major release of
+   RPC OpenStack?)
+
+Development Workflow
+--------------------
+
+#. A bug is filed or a feature is proposed to |rpcos|
+
+#. After triage and prioritization, a developer creates a fix.
+
+#. The fix is pushed to a branch on that developer's fork.
+
+   This prevents another developer from being able to collaborate on that
+   change.
+
+#. The developer submits a pull request against ``master``. It's reviewed,
+   iterated upon, and eventually merged.
+
+#. If that PR is elligible to be backported, then the associated bug is
+   labeled as such and PRs are submitted to the stable branches.
+
+#. Stable PRs are then reviewed and merged.
+
+#. A developer will (eventually) close the original issue.
+
+Release Workflow
+----------------
+
+.. note::
+    This section is a bit unclear to me even now. It will need refinement.
+
+#. When targetted features, bugs, etc. are merged to the appropriate branch, a
+   Release Candidate (RC) tag is created, e.g., ``r11.0.0rc1``.
+
+#. Our QE and Support teams use that tag to test certain scenarios.
+
+#. We receive feedback and bug reports which are fixed. We then go back to
+   Step 1 until we have a version with which everyone is the least unhappy.
+   The most recent RC tag is also then the final tag, e.g., ``r11.0.0``.
+
+Our official release document (hidden away in a private wiki) states we should
+create a branch for a release before it is created, but recently when we
+released ``r11.1.0``, we created a separate branch ``kilo-11.1`` after we
+decided on the final acceptable ``r11.1.0`` RC tag.
+
+.. _proposed-workflow:
+
+RPC OpenStack's Proposed New Workflow
+=====================================
+
+Pre-Workflow
+------------
+
+#. Rename ``master`` to be |lsbnext|.
+
+#. Make |lsbnext| the "default" branch on GitHub.
+
+   This assures that all PRs will be targetted to |lsbnext| by default.
+
+.. note::
+
+    When a new stable release of OpenStack is created, |lsbnext| would then be
+    renamed to the codename of that release, e.g., ``liberty-next``.
+
```

Comment:
Explicit is far better than implicit. `master` is a meaningless, generic term. `blah-next` tells people the branch's purpose better than `master`. We've had a fair number of people confused about what `master` was since we released `r11.0.0`. Ideally this naming change makes the repository more self-explanatory.